t/core/request.t

```diff
@@ -439,3 +439,59 @@ while ($i <= 101) {
 $s
 --- response_body
 101
+
+
+
+=== TEST 15: add header
+--- config
+    location /t {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            ngx.ctx.api_ctx = {}
+            local ctx = ngx.ctx.api_ctx
+            core.request.add_header(ctx, "test_header", "test")
+            local h = core.request.header(ctx, "test_header")
+            ngx.say(h)
+        }
+    }
+--- response_body
+test
+
+
+
+=== TEST 16: test invalidated cache for add_header
+--- config
+    location /t {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            ngx.ctx.api_ctx = {}
+            local ctx = ngx.ctx.api_ctx
+            core.request.add_header(ctx, "test_header", "test")
+            local h = core.request.header(ctx, "test_header")
+            ngx.say(ctx.headers.test_header)
+            core.request.add_header(ctx, "test_header", "abc")
+            ngx.say(ctx.headers.test_header)
+        }
+    }
+--- response_body
+test
+nil
```

Comment:
Should be `abc` ？

Reply:
`add_header()` method will invalidate cache in `ctx`, `header()` method will add cache to `ctx`, so first value is "test" and the second value is "nil"

Reply:
Why does `add_header()` invalidate the cache?

Reply:
This is the feature of current PR. `header(ctx, header_name)` method maybe get wrong values from ctx cache if the `add_header` method does not invalidate the cache.